src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -188,9 +200,8 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
                     continue;
                 }
 
-                // if the address changed, remember the contact we want to add a new address for    
-                Contact conOld = (listOld != null ? (Contact)listOld[i] : null);
-                if (isContactAddressChanged(con, conOld)) 
+                // if the contacts address changed or if the contact with an address moves to a new household remember the contact
+                if (isContactAddressChanged(con, conOld) || (acctChanged && !isContactAddressEmpty(con))) 
```

Comment:
@jjbennett I have to admit I'm troubled that this code might be adding a contact to listConCreateAddr twice.  If the contact gets added in 158, due to acct changing, not an override, and no address fields changed, won't the contact get added again in 204 is acctChanged and not empty address fields?